rkt_test.go is failing

Array equality test:
```
--- FAIL: TestGetPods (0.04s)
	<autogenerated>:31: 
                        
	Error Trace:	rkt_test.go:502
		
	Error:		Not equal: []*container.Pod{(*container.Pod)(0xc820011f90), (*container.Pod)(0xc82037e000)} (expected)
			        != []*container.Pod{(*container.Pod)(0xc82037e0f0), (*container.Pod)(0xc82037e0a0)} (actual)
			
			Diff:
			--- Expected
			+++ Actual
			@@ -1,3 +1,42 @@
			 ([]*container.Pod) (len=2 cap=2) {
			- (*container.Pod)(0xc820011f90)({
			+ (*container.Pod)(0xc82037e0f0)({
			+  ID: (types.UID) (len=2) "43",
			+  Name: (string) (len=9) "guestbook",
			+  Namespace: (string) (len=7) "default",
			+  Containers: ([]*container.Container) (len=4 cap=4) {
			+   (*container.Container)(0xc820377aa0)({
			+    ID: (container.ContainerID) rkt://uuid-4003:app-11,
			+    Name: (string) (len=6) "app-11",
			+    Image: (string) (len=11) "img-name-11",
			+    Hash: (uint64) 10011,
			+    Created: (int64) 90000,
			+    State: (container.ContainerState) (len=6) "exited"
			+   }),
			+   (*container.Container)(0xc820377b00)({
			+    ID: (container.ContainerID) rkt://uuid-4003:app-22,
			+    Name: (string) (len=6) "app-22",
			+    Image: (string) (len=11) "img-name-22",
			+    Hash: (uint64) 10022,
			+    Created: (int64) 90000,
			+    State: (container.ContainerState) (len=6) "exited"
			+   }),
			+   (*container.Container)(0xc820377b60)({
			+    ID: (container.ContainerID) rkt://uuid-4004:app-11,
			+    Name: (string) (len=6) "app-11",
			+    Image: (string) (len=11) "img-name-11",
			+    Hash: (uint64) 10011,
			+    Created: (int64) 100000,
			+    State: (container.ContainerState) (len=7) "running"
			+   }),
			+   (*container.Container)(0xc820377bc0)({
			+    ID: (container.ContainerID) rkt://uuid-4004:app-22,
			+    Name: (string) (len=6) "app-22",
			+    Image: (string) (len=11) "img-name-22",
			+    Hash: (uint64) 10022,
			+    Created: (int64) 100000,
			+    State: (container.ContainerState) (len=7) "running"
			+   })
			+  }
			+ }),
			+ (*container.Pod)(0xc82037e0a0)({
			   ID: (types.UID) (len=2) "42",
			@@ -6,3 +45,3 @@
			   Containers: ([]*container.Container) (len=2 cap=2) {
			-   (*container.Container)(0xc8203776e0)({
			+   (*container.Container)(0xc8203779e0)({
			     ID: (container.ContainerID) rkt://uuid-4002:app-1,
			@@ -14,3 +53,3 @@
			    }),
			-   (*container.Container)(0xc820377740)({
			+   (*container.Container)(0xc820377a40)({
			     ID: (container.ContainerID) rkt://uuid-4002:app-2,
			@@ -23,41 +62,2 @@
			   }
			- }),
			- (*container.Pod)(0xc82037e000)({
			-  ID: (types.UID) (len=2) "43",
			-  Name: (string) (len=9) "guestbook",
			-  Namespace: (string) (len=7) "default",
			-  Containers: ([]*container.Container) (len=4 cap=4) {
			-   (*container.Container)(0xc8203777a0)({
			-    ID: (container.ContainerID) rkt://uuid-4003:app-11,
			-    Name: (string) (len=6) "app-11",
			-    Image: (string) (len=11) "img-name-11",
			-    Hash: (uint64) 10011,
			-    Created: (int64) 90000,
			-    State: (container.ContainerState) (len=6) "exited"
			-   }),
			-   (*container.Container)(0xc820377800)({
			-    ID: (container.ContainerID) rkt://uuid-4003:app-22,
			-    Name: (string) (len=6) "app-22",
			-    Image: (string) (len=11) "img-name-22",
			-    Hash: (uint64) 10022,
			-    Created: (int64) 90000,
			-    State: (container.ContainerState) (len=6) "exited"
			-   }),
			-   (*container.Container)(0xc820377860)({
			-    ID: (container.ContainerID) rkt://uuid-4004:app-11,
			-    Name: (string) (len=6) "app-11",
			-    Image: (string) (len=11) "img-name-11",
			-    Hash: (uint64) 10011,
			-    Created: (int64) 100000,
			-    State: (container.ContainerState) (len=7) "running"
			-   }),
			-   (*container.Container)(0xc8203778c0)({
			-    ID: (container.ContainerID) rkt://uuid-4004:app-22,
			-    Name: (string) (len=6) "app-22",
			-    Image: (string) (len=11) "img-name-22",
			-    Hash: (uint64) 10022,
			-    Created: (int64) 100000,
			-    State: (container.ContainerState) (len=7) "running"
			-   })
			-  }
			  })
		
	Messages:	test case #2
		
```
kubernetes-jenkins/pr-logs/pull/22853/kubernetes-pull-test-unit-integration/18143

Triggered in #22853 and #22851

cc @ncdc @yifan-gu

--- a/pkg/kubelet/rkt/rkt.go
+++ b/pkg/kubelet/rkt/rkt.go
@@ -998,6 +998,7 @@ func (r *Runtime) GetPods(all bool) ([]*kubecontainer.Pod, error) {
 	}
 
 	pods := make(map[types.UID]*kubecontainer.Pod)
+	var podIDs []types.UID
 	for _, pod := range listResp.Pods {
 		pod, err := r.convertRktPod(pod)
 		if err != nil {
@@ -1009,16 +1010,17 @@ func (r *Runtime) GetPods(all bool) ([]*kubecontainer.Pod, error) {
 		oldPod, found := pods[pod.ID]
 		if !found {
 			pods[pod.ID] = pod
+			podIDs = append(podIDs, pod.ID)
 			continue
 		}
 
 		oldPod.Containers = append(oldPod.Containers, pod.Containers...)
 	}
 
-	// Convert map to list.
+	// Convert map to list, using the consistent order from the podIDs array.
 	var result []*kubecontainer.Pod
-	for _, p := range pods {
-		result = append(result, p)
+	for _, id := range podIDs {
+		result = append(result, pods[id])
 	}
 
 	return result, nil

